lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -380,9 +380,11 @@ public function parse()
      * declarations (from ... x join ... y join ... z ...) appear in the query
      * as the hydration process relies on that order for proper operation.
      *
-     * @param AST\SelectStatement|AST\DeleteStatement|AST\UpdateStatement $AST
+     * @param AST\SelectStatement $AST
+     *
+     * @return void
      */
-    private function fixIdentificationVariableOrder($AST)
+    private function fixIdentificationVariableOrder(AST\SelectStatement $AST)
```

Comment:
This one introduced some test failures, it must accept all three AST statement types as previously specified.